exercises/parallel-letter-frequency/src/test/java/ParallelLetterFrequencyTest.java

```diff
@@ -0,0 +1,216 @@
+            {
+                put((int) 'a', 1);
+            }
+        };
+        ParallelLetterFrequency p = new ParallelLetterFrequency(input);
+
+        assertEquals(expectedOutput, p.letterCounts());
+    }
+
+    @Test
+    @Ignore
+    public void resultsAreCaseInsensitive() {
+        String input = "Aa";
+        Map<Integer, Integer> expectedOutput = new HashMap<Integer, Integer>() {
+            {
+                put((int) 'a', 2);
+            }
+        };
+        ParallelLetterFrequency p = new ParallelLetterFrequency(input);
+
+        assertEquals(expectedOutput, p.letterCounts());
+    }
+
+
+    @Test
+    @Ignore
+    public void biggerEmptyTextsStillReturnNoResults() {
+        StringBuilder b = new StringBuilder();
+        for (int i = 0; i < 10000; i++) {
+            b.append(" ");
+        }
+
+        Map<Integer, Integer> expectedOutput = new HashMap<Integer, Integer>();
+        ParallelLetterFrequency p = new ParallelLetterFrequency(b.toString());
+
+        assertEquals(expectedOutput, p.letterCounts());
+    }
+
+    @Test
+    @Ignore
+    public void manyRepetitionsOfTheSameTextGiveAPredictableResult() {
+        StringBuilder b = new StringBuilder();
+        for (int i = 0; i < 10000; i++) {
+            b.append("abc");
+        }
+
+        Map<Integer, Integer> expectedOutput = new HashMap<Integer, Integer>() {
+            {
+                put((int) 'a', 10000);
+                put((int) 'b', 10000);
+                put((int) 'c', 10000);
+            }
+        };
+        ParallelLetterFrequency p = new ParallelLetterFrequency(b.toString());
+
+        assertEquals(expectedOutput, p.letterCounts());
+    }
+
+
+    @Test
+    @Ignore
+    public void punctuationDoesntCount() {
+        ParallelLetterFrequency p = new ParallelLetterFrequency(OdeAnDieFreude);
+
+        assertFalse(p.letterCounts().containsKey((int) ','));
+    }
+
+    @Test
+    @Ignore
+    public void numbersDontCount() {
+        ParallelLetterFrequency p = new ParallelLetterFrequency("Testing, 1, 2, 3");
+
+        assertFalse(p.letterCounts().containsKey((int) '1'));
+    }
+
+    @Test
+    @Ignore
+    public void allThreeAnthemsTogetherProduceCorrectCounts() {
+        StringBuilder b = new StringBuilder();
+        b.append(OdeAnDieFreude);
+        b.append(Wilhelmus);
+        b.append(StarSpangledBanner);
+
+        ParallelLetterFrequency p = new ParallelLetterFrequency(b.toString());
+
+        assertEquals(new Integer(49), p.letterCounts().get((int) 'a'));
+        assertEquals(new Integer(56), p.letterCounts().get((int) 't'));
+        assertEquals(new Integer(2), p.letterCounts().get((int) 'ü'));
+    }
+
+    @Test
+    @Ignore
+    public void tenThreadPoolIsFasterThanSingleThreadPool()
+            throws InterruptedException, ExecutionException {
+
+        ForkJoinPool singleThreadPool = new ForkJoinPool(1);
+        ForkJoinPool tenThreadPool = new ForkJoinPool(10);
```

Comment:
I'm not super-familiar with `ForkJoinPool`. Does the mechanism we're using for testing place any implicit constraints on how solutions must be structured in order to leverage the multiple threads? If so, that might be something to call out in the hints file for this exercise.